### PR TITLE
feature: S3C-4345

### PR DESCRIPTION
**Pull request by miniscruff**
Add prom-client and metrics for queue populator

### CODE REVIEW

File: lib/queuePopulator/LogReader.js [hunk 1]
```diff
@@ -423,16 +428,29 @@ class LogReader {
...
+                        {
+                            method: 'LogReader._processPublishEntries',
+                            topic,
+                            entryCount: topicEntries.length,
+                            error: err,
+                        }
+                    );
+                    const labels = Object.assign({},
+                        this.getMetricLabels(),
+                        { publishStatus: 'FAILURE' }
```
Comment (naren-scality) [inline lib/queuePopulator/LogReader.js:440]: Do we have constants describing the status, if yes, we should use them instead!

Comment (naren-rajendran) [inline lib/queuePopulator/LogReader.js:440]: Do we need constants for the status?

Comment (miniscruff) [reply lib/queuePopulator/LogReader.js:440]: When we add status processor status values we should standardize terms, for now I think it is fine.

File: lib/queuePopulator/QueuePopulator.js [hunk 2]
```diff
@@ -11,6 +9,102 @@ const MetricsProducer = require('../MetricsProducer');
...
+const logSizeMetric = new promClient.Gauge({
+    name: 'replication_log_size',
+    help: 'Current size of metadata journal',
+    labelNames: metricLabels,
+});
+
+const messageMetrics = new promClient.Counter({
+    name: 'replication_populator_messages',
+    help: 'Total number of Kafka messages produced by the queue populator',
+    labelNames: ['origin', 'logName', 'logId', 'publishStatus'],
```
Comment (naren-rajendran) [inline lib/queuePopulator/QueuePopulator.js]: Don't we need containerName here?

File: lib/queuePopulator/QueuePopulator.js [hunk 3]
```diff
@@ -11,6 +9,102 @@ const MetricsProducer = require('../MetricsProducer');
...
+/**
+ * Contains methods to incrememt different metrics
+ * @typedef {Object} MetricsHandler
+ * @property {CounterInc} messages - Increments the message metric
+ * @property {CounterInc} objects - Increments the objects metric
+ * @property {CounterInc} bytes - Increments the bytes metric
+ * @property {GaugeSet} logReadOffset - Set the log read offset metric
+ * @property {GaugeSet} logSize - Set the log size metric
+ */
+const metricsHandler = {
```
Comment (naren-rajendran) [inline lib/queuePopulator/QueuePopulator.js:101]: Should metricsHandler be a class?

**Comment by naren-rajendran** [comment]
Tests? and a minor comment on metricsHandler, otherwise LGTM.

File: lib/queuePopulator/QueuePopulator.js (hunk 3 above)
Comment (miniscruff) [reply lib/queuePopulator/QueuePopulator.js:101]: Metrics handler is just a collection of metric methods, there is no internal state or parameters to pass so I feel a class is unnecessary.

Comment (naren-rajendran) [reply lib/queuePopulator/QueuePopulator.js:101]: are the functions reusable for processor, we dont want to duplicate!

Comment (miniscruff) [reply lib/queuePopulator/QueuePopulator.js:101]: the GaugeSet and CounterInc methods and types will be for sure, maybe a metrics util file will come in handy. metricsHandler itself will need to be per process but it should be relatively similar across the board.

Comment (naren-rajendran) [reply lib/queuePopulator/QueuePopulator.js:101]: Yes, I think a metrics util should be in place.

**Comment by miniscruff** [comment]
ping

File: lib/tasks/BackbeatTask.js [hunk 4]
```diff
@@ -31,33 +31,11 @@ class BackbeatTask {
         const startTime = Date.now();
         const self = this;
 
-        // FIXME workaround for S3C-4457:
```
Comment (naren-rajendran) [inline lib/tasks/BackbeatTask.js]: Is this right? is the block really removed?

File: tests/unit/lib/tasks/BackbeatTask.spec.js [hunk 5]
```diff
@@ -1,44 +0,0 @@
-const assert = require('assert');
```
Comment (naren-rajendran) [inline tests/unit/lib/tasks/BackbeatTask.spec.js]: Same here, why is the test removed?

File: lib/tasks/BackbeatTask.js (hunk 4 above)
Comment (miniscruff) [reply lib/tasks/BackbeatTask.js]: uhh.... well.... that was not supposed to happen. Let me try this again...

**Review by naren-rajendran — APPROVED** [review]

**Review by rahulreddy — APPROVED** [review]

**Pull request merged**